### PR TITLE
👌 (WOW!) ANTINUKE LOOTGURGLE UPDATE (HUGE!) 👌 (also finished)

### PR DESCRIPTION
**Pull request by Verkister**
# **ESSENTIALLY 3 FANCY NEW GURGLEMODES**
**-Normal digestion has been changed into loot drop system, which gradually gurgles the loot after the prey has ascended to pudge.**
**-An item-friendly variant, which only gurgles the prey's snack stashes. It also ruins their unprotected IDs and contaminates their clothing. (At least wash them first you nasties)**
**-A strip digestion mode, which gradually gurgles the gear off your prey, essentially doing what the main gurglemode does without harming the prey.**
**-Also borgo sleeper gurgles now make the prey drop their posibrains and any subprey within their guts if applicable and they can no longer accidentally eat their own equipment modules off the hud.**

**More details tl;dr chronological order changelog below.**

-Managed to build a vore organ equivalent for the inventory gurgle tweaks I made for dogborg guts.
-This means that gurgled prey now drops their loot into the gut.
-After this, the loot in the gut will slowly gurgle away one item at a time, each giving a trace amount of nourishment, multiplied by the item size.
-All the loot will show up in vore panel. However you need to be bashing refresh to see the progress in real time.
-The forbidden/important items list should still work.
-Not so sure about televore beacons tho. Better keep those in with hold mode and pop em out when you go for gurgles just in case :v 
-Also added in a fix for borgs eating their own modules.
<s>-Slight on-site edits to prevent pocket gear from dropping off its invisible pocket storage holders when dumping contents.</s>
-Gurgled items no longer drop their invisible pocket space holder thingies while dumping their contents.
-IDs and probably some other special items shouldn't randomly vanish now.
-Transfering contents from gut to another now also transfers their preserved status and cleans up the list entries from source gut.
-Ejection, single and complete, now properly clears the lists from ejected stuff.
-Borgo-gurgling a pred should now dump their prey into the borgogut.
-Some other cleanups yadda yadda.
<s>-Added a commented example part hinting that the system is pretty much ready for a trait based item-weak gurgle option.</s>
-All should be well now <s>unless you try to gurgle a synth expecting a preserved brain on borgo gut right now. I just remembered I forgot to steal the posibrain yankage bits from the voreorgan code. Oh well it's not like you could've done that with the old code either :v</s> (fixed)
-Normal itemgurglemode now also takes partial nutritional reagent contents from prey's snack stashes.
-Slightly reduced borgo trash nom sound volume.
-Added an item friendly gurglemode, in which:
-Gurgled prey drops all their loot, but the gut will still act on the stuff over time one by one.
-Instead of gurglin the items, it only gurgles foodstuffs. (with just 0.3x nut/reagent gain to avoid unintentional overdosing from the prey's snack stashes)
-However once picked for handling by the gut, the nonfood items get contaminated if applicable. You probably shouldn't wear/use the gear someone just got digested in unwashed :v
-Added a strip digest mode, that only gurgles the prey's gear instead.
-Various fixes and tweaks to earlier commits.
-Item/subitem handling is now the gut's job. Dying to gurgles still makes the prey drop their outer gear.
-in other words the initial gurgle loot drop now only drops the worn gear, leaving the contents for gut subitem stuffs.
-Added another exclusion after noticing tesh abilities hud thingy in the gut :v
-I swear there was something more.

### CODE REVIEW

**Comment by SpadesNeil** [comment]
I need to make NPC vore spit up items at some point.

**Comment by Verkister** [comment]
Also finished unless a second opinion says it's not.

**Comment by Verkister** [comment]
The last 3 commits were done on-site and untested because I'm too tired for local branch fuckery right now aaaa

**Comment by Verkister** [comment]
tweaking, testing, and polishing in progress...

**Comment by Verkister** [comment]
Finally done 👌 
Tested and working.

**Comment by kcin2001** [comment]
i want this

**Comment by Ketrai** [comment]
This is amazing, good job!

File: code/modules/vore/eating/bellymodes_vr.dm [hunk 1]
```diff
@@ -77,6 +78,188 @@
...
+									internal_contents -= SSF
+									qdel(SSF)
+					if(istype(T, /obj/item/weapon/reagent_containers/food/snacks)) // Weakgurgles still act on foodstuff. Hopefully your prey didn't load their bag with donk boxes.
+						var/obj/item/weapon/reagent_containers/food/snacks/F = T
+						F.reagents.trans_to_mob(owner, (F.reagents.total_volume * 0.3), CHEM_INGEST)
+						internal_contents -= F
+						qdel(F)
+					else
+						items_preserved += T
+						T.contaminate() // Someone got gurgled in this crap. You wouldn't wear/use it unwashed. :v
```
Comment (Cameron653) [inline code/modules/vore/eating/bellymodes_vr.dm]: "You wouldn't wear/use it unwashed"
I wouldn't put it past people.

Everything looks good codewise, though.

**Comment by SpadesNeil** [comment]
@Cameron653 Leaving this to you for review.

**Comment by Cameron653** [comment]
Alright. Codewise it all looks good. Going to download the code and test it to see if it works properly and look for bugs.

**Comment by Cameron653** [comment]
Now that that bug is fixed, I don't believe anything else is wrong here. Ready for merge, I believe.
